src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -88,103 +128,1169 @@ static DamageTable D_80A31B1C = {
+    this->actor.scale.z = 0.0132f;
+    this->actionFunc = EnSlime_UpdateJumpLanding;
+    this->actor.speed = 0.0f;
+}
+
+/**
+ * Update scale/rot for ooze jiggle effect (XZ), update Y scale
+ * to make it appear as though chuchu is compressing.
+ * If > 12 frames have passed since jump start, check again if player
+ * is in range and restart engage/jump cycle if so.
+ * If too far from home, initialize return home cycle (takes priority over
+ * player engage check).
+ * Otherwise, if timer is up, return to idle.
+ */
+void EnSlime_UpdateJumpLanding(EnSlime* this, PlayState* play) {
+    f32 factorScaleY;
+    f32 factorRotXZ;
+    f32 factorScaleXZ;
+
+    EnSlime_UpdateBlink(this);
+    this->timer--;
+    factorScaleY = ((this->timer / 5) + 1) * 1.6f;
+    factorRotXZ = sqrtf(this->timer) * 0.2f;
+    factorScaleXZ = ((cos_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.05f * factorScaleY)) + 1.0f) * 0.01f;
+    this->actor.scale.x = factorScaleXZ;
+    this->actor.scale.z = factorScaleXZ;
+    if (this->timer < 15) {
+        this->actor.scale.y = (1.0f - (sin_rad(this->timer * (2.0f * M_PI / 5.0f)) * (0.04f * factorScaleY))) * 0.01f;
+    }
+    this->actor.shape.rot.x = randPlusMinusPoint5Scaled(0x200) * factorRotXZ;
+    this->actor.shape.rot.z = randPlusMinusPoint5Scaled(0x200) * factorRotXZ;
+
+    if (Actor_WorldDistXZToPoint(&this->actor, &this->actor.home.pos) > this->distLimit) {
+        EnSlime_ReturnHome(this);
+    } else if ((Player_GetMask(play) != PLAYER_MASK_STONE) && (this->actor.xzDistToPlayer < 280.0f) &&
+               (this->timer < 12)) {
+        EnSlime_EngagePlayer(this);
+    } else if (this->timer == 0) {
+        EnSlime_ReturnToIdle(this);
+    }
+}
+
+/**
+ * Initialize reaction to receiving a blunt hit (such as with Goron Punch etc.)
+ * Chuchu does not take damage, but appears slide backwards.
+ * Action function set to UpdateBluntHitReaction.
+ */
+void EnSlime_ReactToBluntHit(EnSlime* this) {
+    this->actor.speed = 10.0f;
+    if (this->actor.velocity.y > 0.0f) {
+        this->actor.velocity.y = 0.0f;
+    }
+    this->collider.base.acFlags &= ~AC_ON;
+    this->timer = 30;
+    Actor_PlaySfx(&this->actor, NA_SE_EN_SLIME_DEFENCE);
+    if (this->collider.base.ac != NULL) {
+        this->actor.world.rot.y = Actor_WorldYawTowardActor(&this->actor, this->collider.base.ac) + 0x8000;
+    } else {
+        this->actor.world.rot.y = this->actor.yawTowardsPlayer + 0x8000;
+    }
+    this->actionFunc = EnSlime_UpdateBluntHitReaction;
+}
+
+/**
+ * Update reaction to receiving a blunt hit (such as with Goron Punch etc.)
+ * Update XZ speed and jiggle effect
+ * AC_ON flag is off until 15 frames of reaction remain.
+ * When timer is up, chuchu will choose whether to engage player, return home,
+ * or return to idle. If chuchu is too far from home, returning home will
+ * take precedence.
+ */
+void EnSlime_UpdateBluntHitReaction(EnSlime* this, PlayState* play) {
+    f32 timerFactor;
```

Comment:
I think this was called `sqrtFactor` in a function above; can you make sure they're consistent?